packages/sdk/src/signature.ts

```diff
@@ -0,0 +1,29 @@
+import { Secp256k1 } from "@iov/crypto";
+import { Encoding } from "@iov/encoding";
+
+import { encodeSecp256k1Pubkey } from "./pubkey";
+import { pubkeyType, StdSignature } from "./types";
+
+export function encodeSecp256k1Signature(pubkey: Uint8Array, signature: Uint8Array): StdSignature {
+  return {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    pub_key: encodeSecp256k1Pubkey(pubkey),
+    // Recovery seems to be unused
+    signature: Encoding.toBase64(Secp256k1.trimRecoveryByte(signature)),
+  };
+}
+
+export function decodeSignature(
+  signature: StdSignature,
+): { readonly pubkey: Uint8Array; readonly signature: Uint8Array } {
+  switch (signature.pub_key.type) {
+    // Note: please don't add cases here without writing additional unit tests
+    case pubkeyType.secp256k1:
+      return {
+        pubkey: Encoding.fromBase64(signature.pub_key.value),
```

Comment:
Minor point, but I want to bring it up as I will unlikely review this code again for a long while:

This decodes from base64 to uint8array. I am wondering about the other inverse...

I assume there is no way to reverse `Secp256k1.trimRecoveryByte(signature)`, and it doesn't matter?

`encodeSecp256k1Pubkey` moves to the compressed form. Do we want to uncompress on decode?

Reply:
Both very good points. Symmetry is such a great tool to verify the design of software.

I think everything that happens inside of @cosmwasm/sdk should be 1. compressed pubkeys and 2. fixed length 64 byte signatures (i.e. no recovery byte). That means all this normalization should happen when an external system (like BCP) wants to connect to the SDK. So the proper fix in my opinion is to remove both conversions and just ensure we receive the correct format.

Reply:
Done in 41ede3605e4f5cc3b11d58be2f2e514a03c7af7d and a11b756ea63ac464db4a0c0d0e858fdc2d27e586